android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageViewModel.kt

```diff
@@ -1,21 +1,40 @@
 package org.uwaterloo.subletr.pages.home
 
+import android.util.Log
+import androidx.lifecycle.viewModelScope
 import androidx.navigation.NavHostController
 import dagger.hilt.android.lifecycle.HiltViewModel
 import io.reactivex.rxjava3.core.Observable
+import kotlinx.coroutines.launch
 import org.uwaterloo.subletr.infrastructure.SubletrViewModel
 import org.uwaterloo.subletr.pages.home.list.HomeListChildViewModel
+import org.uwaterloo.subletr.pages.home.map.HomeMapChildViewModel
+import org.uwaterloo.subletr.services.ILocationService
 import org.uwaterloo.subletr.services.INavigationService
 import javax.inject.Inject
 
 @HiltViewModel
 class HomePageViewModel @Inject constructor(
 	private val navigationService: INavigationService,
-	val homeListViewModel: HomeListChildViewModel,
-) : SubletrViewModel<HomePageUiState>(homeListViewModel) {
+	val homeListChildViewModel: HomeListChildViewModel,
+	val homeMapChildViewModel: HomeMapChildViewModel,
+	val locationService: ILocationService,
+) : SubletrViewModel<HomePageUiState>(
+	homeListChildViewModel,
+	homeMapChildViewModel,
+) {
 	val navHostController: NavHostController get() = navigationService.getNavHostController()
 
-	override val uiStateStream: Observable<HomePageUiState> = homeListViewModel.uiStateStream.map {
+	override val uiStateStream: Observable<HomePageUiState> = homeListChildViewModel.uiStateStream.map {
 		it
 	}
+
+	fun getLocation() {
+		viewModelScope.launch {
+			Log.d(
+				"meme",
+				locationService.getLocation().toString(),
```

Comment:
no

Reply:
I disagree with the "meme" tag but I will be keeping the logs for now.